### ISSUE TITLE
Full Recalculation CMDT via App hitting future calls error

### ISSUE THREAD

**Issue by AmadeusBleac** [jamessimone/apex-rollup#355]
Dear @jamessimone,

i am trying to get the initial full recalculation running so we can use the rollup but unfortunately am hitting the future calls:51 error on the `RollupAsyncProcessor `message, we already scaled down the batch to 100 respectively 1000 and still getting the issue.

`Rollup v1.5.19: finalizer logging failure from:
System.LimitException: Too many future calls: 51
Trigger.notifyBackendDealer: line 4, column 1
Inner stacktrace: External entry point`

can you help out here? 

p.s. I am not a dev so please be patient.

**Comment by jamessimone** [comment]
@AmadeusBleac Apex Rollup doesn't make any `@future` calls - it sounds like you may have a bulkification issue in your `Trigger` class within the `notifyBackendDealer` method. In layman's terms, this may help:

* go to Settings -> Custom Metadata Types -> Click "Manage Records" on Rollup Control -> modify the following two properties:
 * `Max Lookup Rows Before Batching` -> change this to 50
 * `Batch Chunk Size` -> change this to something 50 or below

That should help until such a time as your `Trigger` class can be refactored to handle multiple records at once.
